tests/ui/traits/new-solver/normalizes_to_ignores_unnormalizable_candidate.rs

```diff
@@ -0,0 +1,39 @@
+// [no_self_infer] check-pass
+// compile-flags: -Ztrait-solver=next
+// revisions: self_infer no_self_infer
+
+// checks that the new solver is smart enough to infer `?0 = U` when solving:
+// `normalizes-to(<Vec<?0> as Trait>::Assoc, u8)`
+// with `normalizes-to(<Vec<U> as Trait>::Assoc, u8)` in the paramenv even when
+// there is a separate `Vec<T>: Trait` bound  in the param that the `?0` could
+// be inferred to `T` from.
+//
+// FIXME(-Ztrait-solver=next)
+// This should also compile for `normalizes-to(<?0 as Trait>::Assoc, u8)` but currently does not.
```

Comment:
```suggestion
// This could also compile for `normalizes-to(<?0 as Trait>::Assoc, u8)` but
// we currently immediately consider a goal ambiguous if the self type is an
// inference variable.
```